podaac/merger/merge_worker.py

```diff
@@ -25,6 +26,37 @@ def shared_memory_size():
         return int(default_memory_size)
 
 
+def max_var_memory(file_list, var_info, max_dims):
+    """
+    function to get the maximum shared memory that will be used for variables
+
+    Parameters
+    ----------
+    file_list : list
+        List of file paths to be processed
+    var_info : dict
+        Dictionary of variable paths and associated VariableInfo
+    """
+
+    max_var_mem = 0
+    for file in file_list:
+        with nc.Dataset(file, 'r') as origin_dataset:
+
+            for var_path, var_meta in var_info.items():
+                ds_group, var_name = resolve_group(origin_dataset, var_path)
+                ds_var = ds_group.variables.get(var_name)
+
+                if ds_var is None:
+                    target_shape = tuple(max_dims[f'/{dim}'] for dim in var_meta.dim_order)
+                    var_size = math.prod(target_shape) * var_meta.datatype.itemsize
+                    max_var_mem = max(var_size, max_var_mem)
```

Comment:
I think it should be a summation here, not a `max` function

Reply:
Our limitation is the shared memory space not the overall memory space, in which case we put a single variable in the shared memory space and want to know if the largest single variable would fit.

Reply:
Ah I see - so the memory limit lock implemented here (https://github.com/podaac/concise/blob/c8ff81964c606c1f6c7befdb2a3bb3f051ef829a/podaac/merger/merge_worker.py#L214) prevents against a overall memory space issue and this is preventing against the case where a single variable is too large

Reply:
Actually both are preventing shared memory overload, it just that were starting to see larger granules with larger variables and now single variables aren't able to fit into the shared memory, what we were doing in code above is limit how much were loading into the shared memory based on how much we already have in it.

Reply:
Gotcha. Yeah, in this case where a single variable would have trouble fitting into memory, the multi-core performance would be worse than the single-core performance anyways